atlas_densities/densities/mtype_densities_from_map/create.py

```diff
@@ -57,88 +51,62 @@ def create_from_probability_map(
+
+    _check_probability_map_consistency(probability_map, set(molecular_type_densities.keys()))
+
+    region_masks = {
+        region_acronym: annotation.raw == region_id
+        for _, region_acronym, region_id in region_info.itertuples()
+    }
+
+    Path(output_dirpath).mkdir(exist_ok=True, parents=True)
+
+    for mtype in tqdm(probability_map.columns):
+
+        coefficients: Dict[str, Dict[str, Any]] = {}
+        for region_acronym in region_acronyms:
+            coefficients[region_acronym] = {
+                molecular_type: probability_map.at[(region_acronym, molecular_type), mtype]
+                for molecular_type in list(molecular_type_densities.keys())
+                if (region_acronym, molecular_type) in probability_map.index
+            }
 
-    layer_masks = get_layer_masks(annotation.raw, region_map, metadata)
-    zero_density_mtypes = []
-    for mtype in probability_map.columns:
         mtype_density = np.zeros(annotation.shape, dtype=float)
-        coefficients = _get_coefficients(
-            mtype, probability_map, layer_names, list(molecular_type_densities.keys())
-        )
-        for layer_name in layer_names[1:]:
-            for molecular_type, density in molecular_type_densities.items():
-                if molecular_type != "gad67" and not (
-                    layer_name == "layer_6" and molecular_type == "vip"
-                ):
-                    mtype_density[layer_masks[layer_name]] += (
-                        density[layer_masks[layer_name]] * coefficients[layer_name][molecular_type]
-                    )
-                elif layer_name == "layer_6" and molecular_type == "vip":
-                    mtype_density[layer_masks[layer_name]] += (
-                        density[layer_masks[layer_name]] * coefficients[layer_name]["approxlamp5"]
-                    )
-
-        density = molecular_type_densities["gad67"]
-        mtype_density[layer_masks["layer_1"]] = (
-            density[layer_masks["layer_1"]] * probability_map.at["layer_1_gad67", mtype]
-        )
-
-        # Saving to file
-        mtype_filename = f"{mtype.replace('_', '-')}_densities.nrrd"
-        (Path(output_dirpath) / "no_layers").mkdir(exist_ok=True, parents=True)
-        if not np.isclose(np.sum(mtype_density), 0.0):
-            filepath = str(Path(output_dirpath) / "no_layers" / mtype_filename)
-            L.info("Saving %s ...", filepath)
+        for region_acronym in region_acronyms:
+            region_mask = region_masks[region_acronym]
+            for molecular_type, coefficient in coefficients[region_acronym].items():
+                if coefficient <= 0.0:
+                    continue
+                density = molecular_type_densities[molecular_type]
+                mtype_density[region_mask] += density[region_mask] * coefficient
+
+        if np.any(mtype_density):
+            mtype_filename = f"{mtype.replace('_', '-')}_densities.nrrd"  # do we need this?
```

Comment:
which part? the separation of of the filename?  we'll have to make a decision about that - we can choose any style filename we want, but what we will need to do is output a json file giving information about the files.

The information will likely come from the column headers (ie: `BP_bAC`) is a concatenation of the mtype `BP` and the mtype` bAC`; however, having a separator of `_` is dangerous, as mtypes (and etypes?) could potentially have that in their name - thus creating uncertainty.
I'm considering suggesting a different separator (` || `) or something, so that we can for sure separate the mtype and etype.  We would then output the filenames however we want, but create a file like:
```
{
 "$mtype0": {
    "$etype0" : "path/to/filename.nrrd",
    "$etype1" : "path/to/other-filename.nrrd",
    },
   "L5_TPC:C": { "cIR": "path/to/L5_TPC_C_cIR.nrrd"}
}
```
That will be the next step.